### ISSUE TITLE
Add `findsites` for infinite MPS

### ISSUE THREAD

**Issue by mtfishman** [ITensor/ITensorInfiniteMPS.jl#33]
```julia
using ITensors
using ITensorInfiniteMPS

ITensorInfiniteMPS.getcell(i::Index) = ITensorInfiniteMPS.getcell(tags(i))
function ITensors.findfirstsiteind(ψ::InfiniteMPS, i::Index)
  c = ITensorInfiniteMPS.getcell(i)
  i1 = translatecell(i, -(c - 1))
  n1 = findfirst(hascommoninds(i1), ψ[Cell(1)])
  return (c - 1) * nsites(ψ) + n1
end
function ITensors.findsites(ψ::InfiniteMPS, T::ITensor)
  s = filterinds(T; plev=0)
  return [ITensors.findfirstsiteind(ψ, i) for i in s]
end
ITensors.findsites(ψ::InfiniteCanonicalMPS, T::ITensor) = findsites(ψ.AL, T)

N = 2
heisenberg_space_shift(q̃nf, q̃sz) = [QN("Sz", 1 - q̃sz) => 1, QN("Sz", -1 - q̃sz) => 1]
heisenberg_space = fill(heisenberg_space_shift(1, 0), N)
s = infsiteinds("S=1/2", N; space=heisenberg_space)
initstate(n) = isodd(n) ? "↑" : "↓"
ψ = InfMPS(s, initstate)
model = Model("heisenberg")
H = InfiniteITensorSum(model, s)

@show findsites(ψ, H[(1, 2)])
@show findsites(ψ, H[(2, 3)])
@show findsites(ψ, H[(3, 4)])
@show findsites(ψ, H[(4, 5)])
```

**Comment by mtfishman** [comment]
@LHerviou maybe this is useful for #31, if so please feel free to add it to that PR or I can make a separate PR with this functionality.

**Comment by LHerviou** [comment]
I included the findsites, with some added functionalities.

**Comment by mtfishman** [comment]
Closed by #31.
